shared/src/main/scala/amf/plugins/document/graph/parser/FlattenedGraphParser.scala

```diff
@@ -493,12 +494,24 @@ class FlattenedGraphParser()(implicit val ctx: GraphParserContext) extends Graph
           val parsed = parseSortedArray(l.element, node.as[YMap])
           instance.setArrayWithoutId(f, parsed, annotations(nodes, sources, key))
         case a: Array =>
-          val items = node.as[Seq[YNode]]
-          val values: Seq[AmfElement] = a.element match {
-            case _: Obj    => items.flatMap(n => parse(n.as[YMap]))
-            case Str | Iri => items.map(n => str(value(a.element, n)))
+          (node.tagType, a.element) match {
```

Comment:
Could be refactored to make it more self-explanatory in why we are parsing arrays from a map?